### PR TITLE
Allow overriding the active plugin list

### PR DESCRIPTION
**Pull request by vojtechszocs**
Follow-up to #1871 (un-merged)

This PR adds the ability to override the list of active Console plugins.

Active plugins are used
- for testing: `packages/console-app/src/__tests__/plugin-test-utils.ts` creates a test registry from all active plugins (for "what's tested" vs. "what's built" consistency)
- for building: `webpack.config.ts` creates a virtual `@console/active-plugins` module that returns the list of all active plugins for use in Console (i.e. plugins to be included in the output bundle)
- for reporting: `plugin-stats.ts` dynamically loads entry modules of all active plugins (executed as part of devel & prod builds)

By default, plugins **must** be listed in `packages/console-app/package.json` file under `dependencies` key in order to be considered active.

With this PR, it's possible to override that via `CONSOLE_PLUGINS` env. variable:

```sh
# use standard active plugin resolution logic
yarn dev

# for core Console development (no additional plugins)
CONSOLE_PLUGINS=@console/app yarn dev

# for customized plugin development experience
CONSOLE_PLUGINS=@console/app,@console/foo,@console/bar yarn dev
```

The `CONSOLE_PLUGINS` override is consistently applied to

- `dev` (webpack)
- `build` (webpack)
- `test` (Jest / extension checks)
- `plugin-stats` (`ts-node`)

/cc @spadgett @alecmerdler @christianvogt

### CODE REVIEW

**Comment by vojtechszocs** [comment]
cc @rawagner @mareklibra @jelkosz

File: frontend/packages/console-plugin-sdk/src/codegen/plugin-resolver.ts [hunk 1]
```diff
@@ -59,7 +67,21 @@ export const resolvePluginPackages = (
 ) => {
   const packageFiles = glob.sync(consolePkgGlob, { cwd: monorepoRootDir, absolute: true });
   const { appPackage, pluginPackages } = readPackages(packageFiles);
-  return appPackage ? pluginFilter(appPackage, pluginPackages) : [];
+
+  if (!appPackage) {
+    throw new Error(`Cannot detect Console application package ${consoleAppName}`);
```
Comment (vojtechszocs) [inline frontend/packages/console-plugin-sdk/src/codegen/plugin-resolver.ts]: Note: this check was added to reflect the expectation that Console (app) itself is a plugin, i.e. `packages/console-app/src/plugin.tsx`.

**Comment by vojtechszocs** [comment]
@alecmerdler @spadgett If you'd like, we can use a shorter form e.g.

```sh
CONSOLE_PLUGINS=app,foo,bar yarn dev
```

instead of

```sh
CONSOLE_PLUGINS=@console/app,@console/foo,@console/bar yarn dev
```

since all Console plugin pkg names are expected to use `@console` scope anyway.

**Comment by christianvogt** [comment]
@vojtechszocs being explicit is fine. You could add support for both and if there's no scope you assume the `@console` scope.

**Comment by vojtechszocs** [comment]
> @vojtechszocs being explicit is fine. You could add support for both and if there's no scope you assume the `@console` scope.

Agreed, will do.

**Comment by alecmerdler** [comment]
Before:
```
$ yarn run dev
...
9525 modules
Time: 52448ms
```
After:
```
$ yarn run dev
...
8594 modules
Time: 44883ms
```
Nice!

File: frontend/__mocks__/localStorage.ts [hunk 2]
```diff
@@ -2,10 +2,10 @@ let _localStorage = {};
 
 (window as any).localStorage = (window as any).sessionStorage = {
   setItem(key, value) {
-    return Object.assign(_localStorage, {[key]: value});
+    Object.assign(_localStorage, { [key]: value} );
```
Comment (alecmerdler) [inline frontend/__mocks__/localStorage.ts]: I guess this _does_ adhere to the [real web API](https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem). But the existing way is what it _should_ be (in a FP world).

Comment (vojtechszocs) [reply frontend/__mocks__/localStorage.ts]: Yup, the web storage API could be better :smiley: the above change just reflects the method signature in `node_modules/typescript/lib/lib.dom.d.ts`

```ts
setItem(key: string, value: string): void;
```

**Pull request merged**